presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -26,6 +26,7 @@
 
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.NotThreadSafe;
```

Comment:
To me it is strange, that caching has to be done here to avoid filter recompilation. Why not to cache filter compilation object directly in other place? 

Please forgive if I just said something stupid.